glossary.rst

```diff
@@ -0,0 +1,46 @@
+.. _glossary:
+
+========
+Glossary
+========
+
+This page contains definitions for Flux-specific terms and concepts mentioned
+throughout the various guides and docs on our sites.
+
+------------------------------------------------------------------------------
+
+**Flux Broker**
+  A distributed message broker daemon that provides communications services
+  within a Flux instance.
+
+**Flux Instance**
+  A group of flux-broker processes which are launched via any parallel launch
+  utility that supports PMI (Process Manager Interface).
+
+**Hostlist**
+  A format supported by existing HPC tools to create a compact list of hosts by
+  name, consisting of a common prefix, a number, and an optional suffix.
+
+**Jobspec**
+  A file which defines the resources, tasks, and other attributes of a single
+  program.
+
+**Jobtap Plugin**
```

Comment:
An example?

Reply:
Hmmm, I wonder if there's an example small enough to paste on this page. I guess I could paste a link to the [flux-accounting jobtap plugin](https://github.com/cmoussa1/flux-accounting/blob/master/src/plugins/mf_priority.cpp) or the examples in the [flux-core repo](https://github.com/flux-framework/flux-core/tree/master/t/job-manager/plugins) (thinking about it out loud, linking the latter might make more sense as those are written in C, like the rest of flux-core). What do you think?

Reply:
I think that would work for now - in the future for developer docs we will want documentation that unwraps what it means to be a plugin, how to write one, etc.

Reply:
there are some simple jobtap plugins used for testing in the flux-core testsuite. However, I'm not convinced copy and paste of some C code would elucidate much here since even a simple plugin might be quite verbose. Would it be enough to paste a link to the `flux-jobtap-plugins(7)` man page?